unit-tests/py/rspy/devices.py

```diff
@@ -132,13 +133,69 @@ def query( monitor_changes = True ):
             else:
                 sn = dev.get_info( rs.camera_info.serial_number )
             device = Device( sn, dev )
+            if device.port is not None:
+                known_ports.append(device.port)
             _device_by_sn[sn] = device
             log.d( '... port {}:'.format( device.port is None and '?' or device.port ), dev )
     finally:
         log.debug_unindent()
     #
     if monitor_changes:
         _context.set_devices_changed_callback( _device_change_callback )
+    if acroname:
+        need_recover = False
+        for dev in _device_by_sn.values():
+            if dev.port is None:
+                need_recover = True
+        if need_recover:
+            log.d( 'trying do discover unknown ports' )
+            ports = acroname.ports()
+            unknown_ports = [port for port in ports if port not in known_ports]
+            print( "ports:", ports )
```

Comment:
Use `log.d` and `debug_indent()` with a heading saying what it is you're doing

Reply:
Done, I forgot, sorry